extensions/machine-learning-services/src/common/utils.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as uuid from 'uuid';
+import * as path from 'path';
+import * as os from 'os';
+import { promises as fs, existsSync } from 'fs';
+import * as constants from '../common/constants';
+
+export async function execCommandOnTempFile<T>(content: string, command: (filePath: string) => Promise<T>): Promise<T> {
+	let tempFilePath: string = '';
+	try {
+		tempFilePath = path.join(os.tmpdir(), `ads_ml_temp_${uuid.v4()}`);
+		await fs.writeFile(tempFilePath, content);
+		let result = await command(tempFilePath);
+		return result;
+	}
+	catch (err) {
+		throw err;
+	}
+	finally {
+		fs.unlink(tempFilePath);
+	}
+}
+
+export async function exists(path: string): Promise<boolean> {
+	return new Promise<boolean>(resolve => {
+		try {
+			// tslint:disable-next-line:no-sync
+			resolve(existsSync(path));
```

Comment:
Instead of this I would suggest doing what Anthony said and just promisifying the non-sync exists. This is still going to have the problem of blocking the thread until it completes as is. [https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/node/pfs.ts#L169](https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/node/pfs.ts#L169)